packages/medusa/src/loaders/search-index.ts

```diff
@@ -1,67 +1,19 @@
-import ProductService from "../services/product"
-import { indexTypes } from "medusa-core-utils"
 import { MedusaContainer } from "../types/global"
 import DefaultSearchService from "../services/search"
 import { Logger } from "../types/global"
+import { EventBusService } from "../services"
 
-async function loadProductsIntoSearchEngine(
-  container: MedusaContainer
-): Promise<void> {
-  const searchService = container.resolve<DefaultSearchService>("searchService")
-  const productService = container.resolve<ProductService>("productService")
-
-  const TAKE = 20
-  let hasMore = true
+export const SEARCH_INDEX_EVENT = "SEARCH_INDEX_EVENT"
 
-  let lastSeenId = ""
-
-  while (hasMore) {
-    const products = await productService.list(
-      { id: { gt: lastSeenId } },
-      {
-        select: [
-          "id",
-          "title",
-          "status",
-          "subtitle",
-          "description",
-          "handle",
-          "is_giftcard",
-          "discountable",
-          "thumbnail",
-          "profile_id",
-          "collection_id",
-          "type_id",
-          "origin_country",
-          "created_at",
-          "updated_at",
-        ],
-        relations: [
-          "variants",
-          "tags",
-          "type",
-          "collection",
-          "variants.prices",
-          "images",
-          "variants.options",
-          "options",
-        ],
-        take: TAKE,
-        order: { id: "ASC" },
-      }
+function loadProductsIntoSearchEngine(container: MedusaContainer): void {
+  const logger: Logger = container.resolve<Logger>("logger")
+  const eventBusService: EventBusService = container.resolve("eventBusService")
+  eventBusService.emit(SEARCH_INDEX_EVENT, {}).catch((err) => {
+    logger.error(err)
+    logger.error(
+      "Something went wrong while indexing the product into the searchedusa engine."
```

Comment:
The catch-block here doesn't catch errors while the event is being handled but only catches a potential error when the EventBusService fails to add the event to the queue. Would change the log here:

```suggestion
  await eventBusService.emit(SEARCH_INDEX_EVENT, {}).catch((err) => {
    logger.error(err)
    logger.error(
      "Something went wrong while emitting the search indexing event."
```